src/Go/effects/netscriptGoImplementation.ts

```diff
@@ -264,6 +264,17 @@ export function getControlledEmptyNodes(_board?: Board) {
   );
 }
 
+export function setTestingBoardState(board: Board) {
+  resetBoardState(
+    () => {},
+    () => {},
+    GoOpponent.none,
+    board.length,
+  );
+  Go.currentGame.board = board;
+  GoEvents.emit();
```

Comment:
I think you need to reset the promises here?

Reply:
resetBoardState() calls the promise resetter added in the recently-merged PR